client/src/main/scala/org/apache/celeborn/client/RequestLocationCallContext.scala

```diff
@@ -36,11 +37,12 @@ trait RequestLocationCallContext {
 
 case class ChangeLocationsCallContext(
     context: RpcCallContext,
-    partitionCount: Int)
+    partitionCount: Int,
+    serdeVersion: SerdeVersion)
   extends RequestLocationCallContext with Logging {
-  val endedMapIds = new util.HashSet[Integer]()
+  val endedMapIds = new util.ArrayList[Integer]()
```

Comment:
Why change HashSet to ArrayList? Is there a specific reason or benefit for doing so?